### ISSUE TITLE
Long Term Plans

### ISSUE THREAD

**Issue by amree** [amree/teecket-web#5]
- [ ] Load result using AJAX
- [ ] Multiple date selection
- [ ] Round trip support
### Load Result Using AJAX
- Bagi user rasa lebih 'laju'
- Response dalam bentuk JSON
- Proses JSON tu dalam page yang sama
- Untuk loading, tak guna library yang guna modal, just loading dalam div tempat result seperti http://tobiasahlin.com/spinkit/
### Multiple Date Selection
- Kena siapkan yang atas dulu
- User boleh pilih lebih dari satu tarikh
- URL akan jadi macam ni: `http://teecket.herokuapp.com/KBR/KUL/13-10-2015,14-10-2015`
- Paparan akan dibuat dalam bentuk sekarang, cuma ada header untuk tarikh, so untuk contoh di atas, akan ada dua table
### Round trip support
- Yang ni patut buat lepas dua-dua yang kat atas selesai
- Akan ada some sort of split dalam page jika guna pergi dan balik
- Kena ada dua field tarikh
- Lebih kurang macam fleapy
- Design belah kiri atau kanan macam sekarang, cuma ada dua sahaja, macam dua kolum
- URL mungkin akan jadi macam ni: `http://teecket.herokuapp.com/KBR/KUL/13-10-2015,14-10-2015/10-10-2015`

**Comment by amree** [comment]
Minta @AmirolAhmad komen

**Comment by AmirolAhmad** [comment]
Yes untuk `Load result using AJAX`, aku setuju sangat.. maybe bleh tambah guna background processing then render the result guna JSON.

`Multiple date selection` instead guna commas kt url, maybe bleh guna '&' sbb xpasti url bleh support commas ke tak. bagus sebab user bleh tgk perbezaa harga untuk dua tarikh yang berbeza.

 `Round trip support` meaning return trip kn? good idea jugak tu.. means kena ada tambahan untok return date..

nak tambah satu kalau boleh select passenger (adult, child n infant)

**Comment by amree** [comment]
Aku ingatkan comma boleh support, nanti aku check, bukan apa, nak bagi nampak url lawa jer, lol

Bab Adult, Child and Infant kena update balik gem teecket. Possible jugak, untuk long term. 

Gem teecket tu hanya boleh guna `Teecket.search` sahaja buat masa ni sebab aku tak sure kenapa tak boleh buat satu-satu macam `AirAsia.new(...).search`, padahal dah public. Kena check balik. Kalau setel yang tu, lebih banyak freedom kat web ni, boleh request dan manipulate mana-mana result yang dikehendaki.

**Comment by AmirolAhmad** [comment]
Awesome.. aku x godek2 lagi repo teecket tu.. nti nk fork n tgk apa yg patut huhu
